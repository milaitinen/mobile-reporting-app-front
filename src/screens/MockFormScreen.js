import React from 'react';
import { View, FlatList, Text, StyleSheet } from 'react-native';

const dummyForms = [...new Array(12)].map((e, i) => `MockForm${i + 1}`);


export default class MockForm extends React.Component {
  static displayName = 'MockForm';

  render() {
      return (
          <View>
              <FlatList
                  data={dummyForms}
                  renderItem={({ item }) => <Text style={styles.FlatListItemStyle}>{item}</Text>}
                  keyExtractor={(item, index) => index}
              />
          </View>
      );
  }
}

const styles = StyleSheet.create({
    FlatListItemStyle: {
        padding: 10,
        fontSize: 15,
        height: 44
    }
});