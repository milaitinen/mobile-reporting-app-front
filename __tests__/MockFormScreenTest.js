import 'react-native';
import React from 'react';
import MockFormScreen from '../src/screens/MockFormScreen';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <MockFormScreen />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});