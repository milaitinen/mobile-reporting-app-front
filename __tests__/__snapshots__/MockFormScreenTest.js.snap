// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RCTScrollView
    data={
      Array [
        "MockForm1",
        "MockForm2",
        "MockForm3",
        "MockForm4",
        "MockForm5",
        "MockForm6",
        "MockForm7",
        "MockForm8",
        "MockForm9",
        "MockForm10",
        "MockForm11",
        "MockForm12",
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm2
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm3
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm4
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm5
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm6
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm7
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm8
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm9
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
              "height": 44,
              "padding": 10,
            }
          }
        >
          MockForm10
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
